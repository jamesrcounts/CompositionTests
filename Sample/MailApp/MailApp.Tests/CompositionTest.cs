using System.ComponentModel.Composition;
using CompositionTests;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace MailApp.Tests
{
    /// <summary>
    /// This example shows one scenario where CompositionInfo needs both catalog and host
    /// to fully analyze composition.
    /// </summary>
    [TestClass]
    public class CompositionTest
    {
        static CompositionTest()
        {
            Program.AddInstance(new TestingSmtpHost());
        }

        [TestMethod]
        public void VerifyComposition()
        {
            MefComposition.VerifyCompositionInfo(Program.Catalog, Program.Host);
        }

        [Export(typeof(SmtpHost))]
        public class TestingSmtpHost : SmtpHost
        {
            public TestingSmtpHost()
                : base("localhost")
            {
            }
        }
    }
}