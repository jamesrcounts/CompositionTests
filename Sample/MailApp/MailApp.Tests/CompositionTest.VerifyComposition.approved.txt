[Part] MailApp.BasicMailSender from: TypeCatalog (Types='MailApp.BasicMailSender, MailApp.DefaultPort').
  [Export] MailApp.BasicMailSender (ContractName="MailApp.BasicMailSender")
  [Import] MailApp.BasicMailSender..ctor (Parameter="host", ContractName="MailApp.SmtpHost")
    [SatisfiedBy] MailApp.Tests.CompositionTest+TestingSmtpHost (ContractName="MailApp.SmtpHost") from: MailApp.Tests.CompositionTest+TestingSmtpHost
  [Import] MailApp.BasicMailSender..ctor (Parameter="port", ContractName="MailApp.Port")
    [SatisfiedBy] MailApp.DefaultPort (ContractName="MailApp.Port") from: MailApp.DefaultPort from: TypeCatalog (Types='MailApp.BasicMailSender, MailApp.DefaultPort').

[Part] MailApp.DefaultPort from: TypeCatalog (Types='MailApp.BasicMailSender, MailApp.DefaultPort').
  [Export] MailApp.DefaultPort (ContractName="MailApp.Port")

